### PR TITLE
Refactor make config independent 

### PR DESCRIPTION
**Pull request by cristianoliveira**

### CODE REVIEW

File: proxy/config.go [hunk 1]
```diff
@@ -52,27 +48,20 @@ func (c *Config) GetService(host string) *Service {
...
+	var mutex = &sync.Mutex{}
 
-//NewService gets the new service.
-func NewService(name, url string) Service {
-	return Service{
-		Name: name,
-		URL:  url,
-	}
+	mutex.Lock()
+	defer mutex.Unlock()
```
Comment (adiclepcea) [inline proxy/config.go]: How come you reverted to mutexes and removed channels?

Comment (adiclepcea) [reply proxy/config.go]: If you initiate a different mutex on every call to LoadServices, then the lock will not block correctly. You should use the same mutex on each lock. This way, the mutex is a local scop variable and it you reenter the same method again, you will reinitiate another mutex. You should declare the mutex at the package level or at the Config structure level.

**Comment by cristianoliveira** [comment]
Hey @adiclepcea Do you have any other thoughts about this change?

**Comment by adiclepcea** [comment]
@cristianoliveira, to me it seems to be ok.

**Pull request merged**